### DISCUSSION TITLE
EmitJSEvent starvation?

### DISCUSSION THREAD

**Discussion by lovettchris** [microsoft/react-native-windows#14003] [Q&A] [answered]
I'm using EmitJSEvent to funnel messages from my grpc C++ code up to my react native view and it works, but if I calll EmitJSEvent too quickly some messages get dropped.  This is bad in my case because the messages represent xbox game pad state, and dropping events is not an option.  Is EmitJSEvent consolidating events of the same type? Can I stop that? Is there a way to force the dispatch of EmitJSEvents to keep up with the rate at which I'm sending them?  Or another way to ask the question, is there a way to increase the size of the dispatch queue? Or is there a way to "wait" for the event to be handled before calling the next EmitJSEvent ?

**Comment by chrisglein** [comment, marked as answer]
I don't think it's intentional that these events would be coalesced. This sounds like a bug. Can you open a bug on the repo that includes more details about your repro?

**Comment by lovettchris** [comment]
Thanks for confirming, sounds like you are saying there is a dispatch queue and no coalescing - so presumably the queue is infinite size or does it start dropping events after some number of pending dispatches? I will try and create a smaller repro.

**Reply by chrisglein** [reply]
I don't believe there's a limit, no. A small repo would be awesome, thank you.

**Reply by lovettchris** [reply]
ok, I need a small react native windows app with one react native windows component connected to TS UI and I'm having a bit of trouble building this from scratch, the empty app works no problem, but when I get to https://microsoft.github.io/react-native-windows/docs/native-modules I get a bit lost.  So if you can hack that up in your sleep it would help.  My current repro is part of a much bigger react native system and I don't want to distract the engineers on that project to help me figure out how to do this small repro... what I'd like to do is have a button on the page that fires off a native C++ method, that then spins up a thread to post back to the typescript UI a bunch of EmitJSEvent calls sending incrementing numbers then have the typescript count how many calls it receives to ensure each increment it receives matches the call count, or something like that.
